### ISSUE TITLE
ubuntu 1804 with zsh shows "sh: 8: [[: not found"

### ISSUE THREAD

**Issue by spin6lock** [jesseduffield/lazygit#57]
Hi, I got "sh: 8: [[: not found" in branches section and top of the window. zsh version zsh 5.4.2 (x86_64-ubuntu-linux-gnu). lazygit is on b80e8111c6f3a1347edd69522aa93bacd6f47af9. How can I fix that error?

**Comment by jesseduffield** [comment]
hmm, it shouldn't be showing that error like that anymore. It looks like you're up to date in terms of the lazygit repo. Could you try running `go install github.com/jesseduffield/lazygit` to build the latest version?

**Comment by SolitudeSF** [comment]
you shouldn't be using `[[` bashism at all. its not posix.

**Comment by ph03** [comment]
Seeing same issue on f8ca0dd with ubuntu 1804 (using fish shell, not zsh, but `bash` doesn't work either).

**Comment by jesseduffield** [comment]
The culprit is this script here which gets your branches in reverse chronological order along with their relative last commit time.
```
set -e
git reflog -n100 --pretty='%cr|%gs' --grep-reflog='checkout: moving' HEAD | {
  seen=":"
  git_dir="$(git rev-parse --git-dir)"
  while read line; do
    date="${line%%|*}"
    branch="${line##* }"
    if ! [[ $seen == *:"${branch}":* ]]; then
      seen="${seen}${branch}:"
      if [ -f "${git_dir}/refs/heads/${branch}" ]; then
        printf "%s\t%s\n" "$date" "$branch"
      fi
    fi
  done \
  | sed 's/ months /m /g' \
  | sed 's/ month /m /g' \
  | sed 's/ days /d /g' \
  | sed 's/ day /d /g' \
  | sed 's/ weeks /w /g' \
  | sed 's/ week /w /g' \
  | sed 's/ hours /h /g' \
  | sed 's/ hour /h /g' \
  | sed 's/ minutes /m /g' \
  | sed 's/ minute /m /g' \
  | sed 's/ seconds /s /g' \
  | sed 's/ second /s /g' \
  | sed 's/ago//g' \
  | tr -d ' '
}
```
It's probably the weakest part of the codebase, not the least because it hasn't been written in go. If anybody can put up a PR that cleans this guy up or refactors it out into just git commands and Go, I would very much appreciate it! I'll spend some time right now seeing if I can refactor it

**Comment by tswsl1989** [comment]
While not a proper fix, the following patch seems to have been enough to get the software up and running for me:
```go
diff --git a/gitcommands.go b/gitcommands.go
index 50c1f2a..63d2ffa 100644
--- a/gitcommands.go
+++ b/gitcommands.go
@@ -116,7 +116,7 @@ func platformShell() (string, string) {
        if runtime.GOOS == "windows" {
                return "cmd", "/c"
        }
-       return "sh", "-c"
+       return "bash", "-c"
 }

 func runDirectCommand(command string) (string, error) {
```

**Comment by perror** [comment]
Beware, the go installer still install the buggy version (I tried a few minutes ago).

And, going to `sh` do improve portability (especially on plate-forms that do not have bash by default), but you have to care about the fact that all your shell commands are POSIX compliant (these `[[` in place of `test` have to be avoided).

**Comment by jesseduffield** [comment]
@perror I just merged the PR to master, that ugly bash script is now out of the codebase. Let me know if there are still any issues

**Comment by perror** [comment]
It works like a charm now! :) Just did a `go get -u lazygit` and it was enough.

**Comment by SolitudeSF** [comment]
you should revert to using `sh` instead of `bash`

**Comment by dawidd6** [comment]
This issue seems to be resolved, tested on 16.04 after this referenced pull request merge.

**Comment by spin6lock** [comment]
problem solved after update to latest version:) Thanks @jesseduffield !
